Framework/PythonInterface/test/python/mantid/_testing/AssertAlmostEqualTest.py

```diff
@@ -33,11 +33,8 @@ def test_rtol(self):
         assert_almost_equal(self.ws1, self.ws3, rtol=0.7)
 
     def test_raises(self):
-        with self.assertRaises(AssertionError):
-            assert_almost_equal(self.ws1, self.ws2)
```

Comment:
This negative test that was removed would have prevented the error above.

Reply:
You are being mislead by the diff (or simply misreading it). The original code was
```python
        with self.assertRaises(AssertionError):
            assert_almost_equal(self.ws1, self.ws2)

        with self.assertRaises(ValueError):
            assert_almost_equal(self.ws1, self.ws2, atol=1, rtol=1)
```
and the new version is
```python
        with self.assertRaises(AssertionError):
            assert_almost_equal(self.ws1, self.ws2)
```
not specifying either is still an error. Specifying both is ok.

Reply:
Here is the entire test suite:
``` python
class AssertAlmostEqualTest(unittest.TestCase):
    @classmethod
    def setUpClass(self):
        ws1 = CreateWorkspace(DataX=[0, 1, 2, 3, 4, 5], DataY=[1, 1, 1, 1, 1, 1])
        ws2 = CreateWorkspace(DataX=[0.09, 1.09, 2.09, 3.09, 4.09, 5.09], DataY=[1, 1, 1, 1, 1, 1])
        ws3 = CreateWorkspace(DataX=[0, 2, 4, 6, 8, 10], DataY=[1, 1, 1, 1, 1, 1])
        self.ws1 = ws1
        self.ws2 = ws2
        self.ws3 = ws3

    def test_simple(self):
        assert_almost_equal(self.ws1, self.ws2, atol=1, rtol=1)

    def test_atol(self):
        # compare (ws1 - ws2) < atol
        assert_almost_equal(self.ws1, self.ws2, atol=0.1)

    def test_rtol(self):
        # compare (ws1 - ws2) / (0.5 * (ws1 + ws2)) < rtol
        assert_almost_equal(self.ws1, self.ws3, rtol=0.7)

    def test_raises(self):
        with self.assertRaises(ValueError):
            assert_almost_equal(self.ws1, self.ws2)
```
The only "negative" test remaining is asserting that a value error is raised, which happens if neither `atol` nor `rtol` are specified.

Reply:
Ran the following in a version of workbench containing this change, which passed
``` python
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

from mantid.testing import assert_almost_equal

ws1 = CreateSingleValuedWorkspace(DataValue=1.0)
ws2 = CreateSingleValuedWorkspace(DataValue=2.0)
assert_almost_equal(ws1, ws2, atol=0.0, rtol=0.0)
print("assert passes")
```

Reply:
Should we just treat all `atol` and `rtol` which are less than  `Property.EMPTY_DBL` equal to `Property.EMPTY_DBL` ? We can catch this special corner case then.

Reply:
`EMPTY_DBL` is a special value. One should only ever check for equality